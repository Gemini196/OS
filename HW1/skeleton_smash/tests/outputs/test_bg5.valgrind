==45022== Memcheck, a memory error detector
==45022== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==45022== Using Valgrind-3.13.0 and LibVEX; rerun with -h for copyright info
==45022== Command: /home/student/Desktop/OS/HW1/skeleton_smash/smash
==45022== Parent PID: 45018
==45022== 
==45022== 
==45022== FILE DESCRIPTORS: 6 open at exit.
==45022== Open file descriptor 1: /home/student/Desktop/OS/HW1/skeleton_smash/tests/outputs/test_bg5.out
==45022==    at 0x54EF957: dup2 (syscall-template.S:78)
==45022==    by 0x112CC9: main (smash.cpp:41)
==45022== 
==45022== Open file descriptor 0:
==45022==    at 0x54EF957: dup2 (syscall-template.S:78)
==45022==    by 0x112C77: main (smash.cpp:33)
==45022== 
==45022== Open file descriptor 5:
==45022==    at 0x54EF927: dup (syscall-template.S:78)
==45022==    by 0x112E17: __static_initialization_and_destruction_0(int, int) (smash.cpp:9)
==45022==    by 0x112E33: _GLOBAL__sub_I_stdout_dp (smash.cpp:57)
==45022==    by 0x112EAC: __libc_csu_init (in /home/student/Desktop/OS/HW1/skeleton_smash/smash)
==45022==    by 0x5400C17: (below main) (libc-start.c:266)
==45022== 
==45022== Open file descriptor 4: /home/student/Desktop/OS/HW1/skeleton_smash/tests/outputs/test_bg5.out
==45022==    at 0x54EF927: dup (syscall-template.S:78)
==45022==    by 0x112E07: __static_initialization_and_destruction_0(int, int) (smash.cpp:8)
==45022==    by 0x112E33: _GLOBAL__sub_I_stdout_dp (smash.cpp:57)
==45022==    by 0x112EAC: __libc_csu_init (in /home/student/Desktop/OS/HW1/skeleton_smash/smash)
==45022==    by 0x5400C17: (below main) (libc-start.c:266)
==45022== 
==45022== Open file descriptor 3: /home/student/Desktop/OS/HW1/skeleton_smash/tests/outputs/test_bg5.valgrind
==45022==    <inherited from parent>
==45022== 
==45022== Open file descriptor 2: /home/student/Desktop/OS/HW1/skeleton_smash/tests/outputs/test_bg5.err
==45022==    <inherited from parent>
==45022== 
==45022== 
==45022== HEAP SUMMARY:
==45022==     in use at exit: 821 bytes in 13 blocks
==45022==   total heap usage: 56 allocs, 43 frees, 85,047 bytes allocated
==45022== 
==45022== 21 bytes in 8 blocks are indirectly lost in loss record 1 of 2
==45022==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==45022==    by 0x10B5B9: _parseCommandLine(char const*, char**) (Commands.cpp:47)
==45022==    by 0x10D7C9: BackgroundCommand::execute() (Commands.cpp:564)
==45022==    by 0x10F690: SmallShell::executeCommand(char const*) (Commands.cpp:971)
==45022==    by 0x112D72: main (smash.cpp:51)
==45022== 
==45022== 821 (800 direct, 21 indirect) bytes in 5 blocks are definitely lost in loss record 2 of 2
==45022==    at 0x4C3289F: operator new[](unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==45022==    by 0x10D781: BackgroundCommand::execute() (Commands.cpp:558)
==45022==    by 0x10F690: SmallShell::executeCommand(char const*) (Commands.cpp:971)
==45022==    by 0x112D72: main (smash.cpp:51)
==45022== 
==45022== LEAK SUMMARY:
==45022==    definitely lost: 800 bytes in 5 blocks
==45022==    indirectly lost: 21 bytes in 8 blocks
==45022==      possibly lost: 0 bytes in 0 blocks
==45022==    still reachable: 0 bytes in 0 blocks
==45022==         suppressed: 0 bytes in 0 blocks
==45022== 
==45022== For counts of detected and suppressed errors, rerun with: -v
==45022== ERROR SUMMARY: 1 errors from 1 contexts (suppressed: 0 from 0)
