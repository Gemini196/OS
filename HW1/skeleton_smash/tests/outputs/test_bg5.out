smash> smash: got ctrl-Z
smash: process 46807 was stopped
smash> smash: got ctrl-Z
smash: process 46873 was stopped
smash> smash: got ctrl-Z
smash: process 46941 was stopped
smash> smash: got ctrl-Z
smash: process 47004 was stopped
smash> smash: got ctrl-Z
smash: process 47022 was stopped
smash> [1] sleep 100 : 46807 2 secs (stopped)
[2] sleep 200 : 46873 2 secs (stopped)
[3] sleep 300 : 46941 1 secs (stopped)
[4] sleep 400 : 47004 1 secs (stopped)
[5] sleep 500 : 47022 0 secs (stopped)
smash> sleep 300 : 46941
smash> sleep 400 : 47004
smash> sleep 200 : 46873
smash> 